doozer/doozerlib/cli/scan_osh.py

```diff
@@ -1,29 +1,289 @@
+        """
+        # Eg: osh-cli find-tasks --regex ose-network-tools-container-v4.15
+        cmd = f"osh-cli find-tasks --regex {brew_package_name}"
+
+        _, result, _ = cmd_gather(cmd)
+
+        finished_tasks = result.strip().split("\n")
+
+        for index, task_id in enumerate(finished_tasks):
+            cmd = f"osh-cli task-info {task_id}"
+            self.runtime.logger.info(f"Running: {cmd}")
+
+            _, result, _ = cmd_gather(cmd)
+
+            if "state_label = CLOSED" in result:
+                try:
+                    previous_scan_id = finished_tasks[index + 1]
+
+                    cmd = f"osh-cli task-info {previous_scan_id}"
+                    self.runtime.logger.info(f"Running: {cmd}")
+                    _, result, _ = cmd_gather(cmd)
+
+                    previous_scan_nvr = yaml.safe_load(result.replace(" =", ":"))["label"]  # returns NVR
+
+                    return task_id, previous_scan_id, previous_scan_nvr
+                except IndexError:
+                    return task_id, None, None
+
+    @staticmethod
+    def check_if_vulnerabilities_exist(task_id: str, nvr: str) -> bool:
+        """
+        ProdSec only wants us to check the scan-results-imp.js file of a scan and see if vulnerabilities are reported.
+        """
+
+        response = requests.get(SCAN_RESULTS_URL_TEMPLATE.format(task_id=task_id, nvr=nvr))
+
+        if len(response.json()["defects"]) > 0:
+            return True
+
+        return False
+
+    def get_distgit_name_from_brew_nvr(self, nvr: str) -> str:
+        """
+            Returns the distgit name from the brew nvr
+
+            :param nvr: Full NVR eg: openshift-enterprise-pod-container-v4.14.0-202310031045.p0.g9ef6de6.assembly.stream
+            """
+        brew_package_name = self.koji_session.getBuild(buildInfo=nvr, strict=True)["package_name"]
+
+        _, output, _ = cmd_gather(f"doozer --disable-gssapi -g {self.runtime.group} "
+                                  f"images:print --short '{{component}}: {{name}}'")
+        result = []
+        for line in output.splitlines():
+            result.append(line.split(": "))
+
+        dict_data = {}
+        for line in result:
+            if line:
+                dict_data[line[0]] = line[1]
+
+        return dict_data[brew_package_name]
+
+    @staticmethod
+    def is_latest_scan_different(latest_scan_id, latest_scan_nvr, previous_scan_id, previous_scan_nvr):
+        latest_scan_response = requests.get(
+            SCAN_RESULTS_URL_TEMPLATE.format(task_id=latest_scan_id, nvr=latest_scan_nvr))
+        previous_scan_response = requests.get(
+            SCAN_RESULTS_URL_TEMPLATE.format(task_id=previous_scan_id, nvr=previous_scan_nvr))
+
+        latest_scan_defects = latest_scan_response.json()["defects"]
+        previous_scan_defects = previous_scan_response.json()["defects"]
+
+        for defect in latest_scan_defects:
```

Comment:
Is defect a string here?

Reply:
Nope, its an `dict` :slightly_smiling_face: 

This is the [sample](https://cov01.lab.eng.brq2.redhat.com/osh/task/361037/log/sriov-network-operator-container-v4.14.0-202311020548.p0.g514df18.assembly.stream/scan-results-imp.js). If you check there, you'll see that its a dict. The aim is to raise a ticket only if the diff changes.

Reply:
@jupierce I hope this way of checking diff is acceptable. Please let me know if I missed something?

Reply:
Thanks, lgtm